src/shared/install.c

```diff
@@ -2282,7 +2282,7 @@ static int install_context_mark_for_removal(
                         else {
                                 log_debug_errno(r, "Unit %s not found, removing name.", i->name);
                                 r = install_changes_add(changes, n_changes, r, i->path ?: i->name, NULL);
-                                if (r < 0)
+                                if (r < 0 && r != -ENOENT) /* Remove leftover symlinks even if the unit is gone */
```

Comment:
hmm, i think this is actually a fuckup caused by f31f10a6207efc9ae9e0b1f73975b5b610914017

i.e. the comment above suggests the intention was always to still delete the name, even if `r` is a failure.

I think the intention of f31f10a6207efc9ae9e0b1f73975b5b610914017 was purely to make ENOMEM errors fatal, but not the other ones, but it did much more than that.

I think we really should fix this properly, i.e. make ENOMEM fatal in these cases, but for all other cases keep doing what we are doing.

@keszybz you did f31f10a6207efc9ae9e0b1f73975b5b610914017 originally, can you confirm (or reject) my understanding of what you were trying to do back then